src/ring_theory/dedekind_domain/adic_valuation.lean

```diff
@@ -333,3 +333,43 @@ instance adic_completion.has_lift_t : has_lift_t K (v.adic_completion K) :=
 def adic_completion_integers : valuation_subring (v.adic_completion K) := valued.v.valuation_subring
 
 end is_dedekind_domain.height_one_spectrum
+
+variables (R K)
+
+lemma adic_completion.is_integer (x : v.adic_completion K) :
+  x ∈ v.adic_completion_integers K ↔ (valued.v x : with_zero (multiplicative ℤ)) ≤ 1 := by refl
+
+/-- The natural inclusion of `R` in `adic_completion`. -/
+def inj_adic_completion_integers' : R → (v.adic_completion K) :=
+λ r, (coe : K → (v.adic_completion K)) (algebra_map R K r)
+
```

Comment:
Do you actually need this given you're also introducing `inj_adic_completion_integers`?
```suggestion
```

Reply:
I probably won't need it, I'll delete it for now.